User ip addr validation and retrieval

jmartin-sul opened the pull request:
- instead of using the x-forwarded-for header as-is, parse it as a comma-delimited string and use the first element of the resulting array as the user's IP address
- use InetAddressValidator for more robust IP address validation
- fix tests accordingly
- bump version

maybe commits 1&2 and 3&4 should be collapsed?  i.e., maybe 653f22b and 3219d0d should be a single commit, and likewise for 37030c2 and 37030c2?  happy to squash if others would prefer that.

@tingulfsen @ndushay

jmartin-sul committed:
stricter user IP address validation
* add commons-validator as a dependency (build.gradle)
* use InetAddressValidator to validate the address.  make sure the address is a "dotted-quad" string. (SulWowza.java)
* fix the affected test (which could no longer ignore IP formatting as relevant to the test). (TestSulWowza.java)

jmartin-sul committed:
user IP address validation: improve logging, improve tests
* fix logging to reflect the improved validation (SulWowza.java)
* fix tests to reflect the improved validation and logging (TestSulWowza.java)

jmartin-sul committed:
add getUserIp helper method to SulWowza.java: parse x-forwarded-for and use the first value in the list instead of the whole string

jmartin-sul committed:
TestSulWowza.java: authorizeSession_acceptsIfAuthorized: test x-forwarded-for with multiple IP addresses

tingulfsen commented on src/edu/stanford/dlss/wowza/SulWowza.java:
```diff
@@ -149,11 +149,28 @@ String getStacksUrl(IApplicationInstance appInstance)
+    // of IP addresses in the "x-forwarded-for" header (as there might be proxies between
+    // the user and wowza).
+    // returns an empty string if it can't parse out an IP address.
+    String getUserIp(java.util.Map<String,String> httpReqHeaders)
```
Is there any reason why you didn't just import java.util.Map, so you could leave out the fully qualified class name here?

jmartin-sul replied:
ah, no, i think i was just copying a return type from the javadocs and being absent-minded about tidying it up.  will fix...

tingulfsen commented:
Other than my rather minor comment about the use of java.util.Map instead of just Map, I'm 👍  on this.

jmartin-sul commented:
ok, suggestion implemented.  i cleaned up two other small things while i was there:
- i dropped the commit with the version bump, since the practice is generally for that to happen directly on master.  i figured it wouldn't be a problem to include it here, but since i'm working on another fix for dlss-wowza (see issue #28), it actually seems like there is a bit of flux in this code base, so it's probably best to follow the usual routine.
- i renamed `shortIp` in one of the tests to the now more accurate `invalidIp` (which i'd previously missed).
